mlflow/pyfunc/__init__.py

```diff
@@ -2617,6 +2621,14 @@ def _save_model_with_loader_module_and_data_path(
         mlflow_model.model_size_bytes = size
     mlflow_model.save(os.path.join(path, MLMODEL_FILE_NAME))
 
+    code_dir_subpath = _validate_infer_and_copy_code_paths(
```

Comment:
can we run this before the first `mlflow_model.save`?

Reply:
we can't, because when inferring, it has to create a child process and load the model (via `mlflow.pyfunc.load`) in the child process. Without the saved `MLmodel` file, loading model doen't work.